api/v1alpha1/gitsync_types.go

```diff
@@ -38,16 +45,42 @@ const (
 	// GitSyncConditionConfigured has the status True when the GitSync
 	// has valid configuration.
 	GitSyncConditionConfigured ConditionType = "Configured"
+
+	SourceTypeHelm      SourceType = "Helm"
+	SourceTypeKustomize SourceType = "Kustomize"
+	SourceTypeRaw       SourceType = "Raw"
 )
 
 // GitSyncSpec defines the desired state of GitSync
 type GitSyncSpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// RepositoryPath lists the Git Repository path to watch
-	RepositoryPath RepositoryPath `json:"repositoryPath"`
+	// Name is a unique name
+	Name string `json:"name"`
```

Comment:
We don't need `Name` anymore as we move these field directly under spec from `RepositoryPath`.

Reply:
I also thought to remove it but it was there in the [defined schema](https://github.com/numaproj-labs/numaplane/issues/90#issuecomment-1967347677) so I had left it, I'll remove it now